src/main/java/org/candlepin/subscriptions/tally/facts/FactNormalizer.java

```diff
@@ -206,9 +206,25 @@ private void normalizeSystemProfileFacts(
       normalizedFacts.setCores(
           hostFacts.getSystemProfileCoresPerSocket() * hostFacts.getSystemProfileSockets());
     }
+    if (hostFacts.getSystemProfileArch().equals("x86_64")
```

Comment:
I would switch the order you're checking for equals in to protect against null pointers.

```
"x86_64".equals(hostFacts.getSystemProfileArch())
```

Same for the next conditional with `virtual`.

@kahowell, do we have any enums for these strings that would be appropriate to use here instead of strings?

Reply:
Hey LIndsey, thanks for the feedback. I''l look around the code base I remember seeing a enum, that I could use. For sure I'll have to create one for the arch